### PR TITLE
[pkg/translator/prw] Use flag to control metrics normalization.

### PR DESCRIPTION
**Pull request by kovrus**
**Description:** <Describe what has changed.>

Use flag to control metrics normalization.  It is more convenient to control metrics name normalization via a flag
  when `pkg/translator/prometheusremotewrite` is used as a third-party dependency.

**Link to tracking Issue:** <Issue number if applicable>
https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/22040

**Testing:** <Describe what testing was performed and which tests were added.>
Exisitng unit tests

### CODE REVIEW

**Review by jpkrohling — APPROVED** [review]
This makes way more sense than the current state!

**Review by Aneurysm9 — CHANGES_REQUESTED** [review]
I believe this needs to be considered in conjunction with https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/21743.

File: pkg/translator/prometheus/normalize_name.go [hunk 1]
```diff
@@ -89,6 +89,11 @@ var normalizeNameGate = featuregate.GlobalRegistry().MustRegister(
 	featuregate.WithRegisterReferenceURL("https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/8950"),
 )
 
+// IsNormalizeNameGateEnabled returns true if conversion to the Prometheus naming convention enabled
+func IsNormalizeNameGateEnabled() bool {
```
[the remark this answers is not in the document]
Comment (kovrus) [reply pkg/translator/prometheus/normalize_name.go]: That makes sense. I've dropped it.

**Comment by kovrus** [comment]
> I believe this needs to be considered in conjunction with https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/21743.

I am not sure about that. This PR is an attempt to drop the dependency of `pkg/translator/prw` on the feature normalize, it is independent of what the default should be or anything else. It is just a code restructuring so this module can be used as a third-party dependency elsewhere without using the feature gate to configure its behavior. I think https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/21743 should not be a blocker here.

**Pull request closed without merging**